src/m365/teams/commands/meeting/meeting-create.ts

```diff
@@ -0,0 +1,206 @@
+      },
+      {
+        option: '-p --participants [participants]'
+      },
+      {
+        option: '--organizerEmail [organizerEmail]'
+      },
+      {
+        option: '-r --recordAutomatically'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.startTime && !validation.isValidISODateTime(args.options.startTime)) {
+          return `'${args.options.startTime}' is not a valid ISO date string for startTime.`;
+        }
+        if (args.options.endTime && !validation.isValidISODateTime(args.options.endTime)) {
+          return `'${args.options.endTime}' is not a valid ISO date string for endTime.`;
+        }
+        if (args.options.startTime && args.options.endTime && new Date(args.options.startTime) >= new Date(args.options.endTime)) {
+          return 'startTime value must be before endTime.';
+        }
+        if (args.options.endTime && !args.options.startTime) {
+          return 'startTime should be specified when endTime is specified.';
+        }
+        if (args.options.participants) {
+          if (args.options.participants.indexOf(',') === -1 && !validation.isValidUserPrincipalName(args.options.participants)) {
+            return `${args.options.participants} contains invalid UPN.`;
+          }
+          const participants = args.options.participants.trim().toLowerCase().split(',').filter(e => e && e !== '');
+          if (!participants || participants.length === 0 || participants.some(e => !validation.isValidUserPrincipalName(e))) {
+            return `${args.options.participants} contains one or more invalid UPN.`;
+          }
+        }
+        if (args.options.organizerEmail && !validation.isValidUserPrincipalName(args.options.organizerEmail)) {
+          return `'${args.options.organizerEmail}' is not a valid email for organizerEmail.`;
+        }
+        return true;
+      }
+    );
+  }
+
+  /**
+   * Executes the command
+   * @param logger Logger instance
+   * @param args Command arguments
+   */
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const isAppOnlyAccessToken = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken)!;
+      if (isAppOnlyAccessToken && !args.options.organizerEmail) {
+        throw `The option 'organizerEmail' is required when creating a meeting using app only permissions`;
+      }
+      const graphBaseUrl = await this.getGraphBaseUrl(args.options);
+      const meeting = await this.createMeeting(logger, graphBaseUrl, args.options);
+      await logger.log(meeting);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  /**
+   * Gets the base MS Graph URL for the request
+   * @param options 
+   * @returns correct MS Graph URL for the request
+   */
+  private async getGraphBaseUrl(options: Options): Promise<string> {
```

Comment:
This may be written a bit shorter and I don't think we need a separate function for it. Let's just add it to the createMeeting function.

Reply:
Ok, I included it to the createMeeting function, I hope that it is ok now.